### ISSUE TITLE
fraud(service): pass version with the networkID

### ISSUE THREAD

**Issue by vgonkivs** [celestiaorg/celestia-node#2512]
Fraud Service requires a networkID parameter from the caller and creates a pub-sub subscription + creates a protocolID. it guarantees that messages from previous versions will not be used by a new handler. But in fact, we missed one thing that messages themselves can be changed and it also requires us to update the version of protocols.

Our protocol versions look like this:
```
* fmt.Sprintf("/%s/fraud-sub/%s/v0.0.1", networkID, fraudType) -> for pubsub
* protocol.ID(fmt.Sprintf("/%s/fraud/v0.0.1", networkID)) -> for stream handler
```

What we can do is pass the network with the respective version from the caller side.

wdyt @Wondertan , @renaynay ?

**Comment by Wondertan** [comment]
We can pass message version to the network-id string, but that doesn't scale. Alternatively, we can introduce versioning to the Proof interface, so that service can construct proper string itself. I believe that's the best option.

**Comment by vgonkivs** [comment]
Agree, that keeping `type/version` makes sense. But does this also mean that potentially one node can handle multiple versions of proof?

**Comment by Wondertan** [comment]
> But does this also mean that potentially one node can handle multiple versions of proof?

Yes, if the user of go-fraud registers the same fraud in two versions

**Comment by Wondertan** [comment]
My original idea was to add the Version method to the Proof interface, but we can also solve this by adding version suffix to the fraud type string.

**Comment by vgonkivs** [comment]
Please note, that `protocol.ID(fmt.Sprintf("/%s/fraud/v0.0.1", networkID))` should be modified then. And `FraudMessageRequest` in proto should be extended with Version + `FraudMessageReSponse.Type` should contain the version.

**Comment by vgonkivs** [comment]
If we'll decide to go this way, then I'd suggest to modify fraud-sub representation to :

`fmt.Sprintf("/%s/fraud-sub/v0.0.1/%s", networkID, fraudType)` -> `arabica-9/fraud-sub/v0.0.1/badencoding/v.0.0.2`

* `fraud-sub/v0.0.1 `-> version of proof service.
* `badencoding/v.0.0.2` -> type and version of the proof itself

We should actually rename `fraud-sub` to `proof-sub` 🤔

**Comment by Wondertan** [comment]
SGTM

**Comment by vgonkivs** [comment]
To avoid any breaking changes in go-fraud library, I decided to introduce a versioning only for the BEFP.
